pytorch_lightning/callbacks/finetuning.py

```diff
@@ -106,11 +144,24 @@ def freeze(module: Module, train_bn: bool = True) -> None:
             train_bn: If True, leave the BatchNorm layers in training mode
         """
         for mod in module.parameters():
-            if (isinstance(mod, BaseFinetuningCallback.BN_TYPES) and train_bn):
-                BaseFinetuningCallback._make_trainable(mod)
+            if (isinstance(mod, BaseFinetuning.BN_TYPES) and train_bn):
+                BaseFinetuning._make_trainable(mod)
             else:
                 mod.requires_grad = False
 
+    @staticmethod
+    def filter_on_optimizer(optimizer, params):
+        out_params = []
+        for param in params:
+            is_in = False
+            for group in optimizer.param_groups:
+                for p in group["params"]:
+                    if p.shape == param.shape and torch.equal(p, param):
+                        is_in = True
+            if not is_in:
+                out_params.append(param)
```

Comment:
```suggestion
            for group in optimizer.param_groups:
                for p in group["params"]:
                    if torch.equal(p, param):
                        break
            else:
                out_params.append(param)
```

changes:
- Use for..else
- No need to check shape and equals